src/Npgsql/NpgsqlTransaction.cs

```diff
@@ -447,7 +453,7 @@ void CheckReady()
         void CheckDisposed()
         {
             if (IsDisposed)
-                throw new ObjectDisposedException(typeof(NpgsqlTransaction).Name);
+                throw new ObjectDisposedException(typeof(NpgsqlTransaction).Name, _disposeReason);
```

Comment:
Note that we don't currently include the break reason in other ObjectDisposedException scenarios (e.g. NpgsqlDataReader), am not sure there's a lot of value in doing this. But I don't have any specific objection - if we think this is valuable we can do it for NpgsqlDataReader and others.

Reply:
It's just a little bit more info whenever a connection is broken (for example by keepalive) without having to enable the logging.

Ideally, we shouldn't throw `ObjectDisposedException` at all in this case (since a user didn't actually disposed anything), but that's a good enough solution.

Reply:
> But I don't have any specific objection - if we think this is valuable we can do it for NpgsqlDataReader and others.

The problem is, we don't dispose `NpgsqlDataReader` on break but close it (I guess it was done like this so `NpgsqlDataReader` will throw a correct exception whenever it's accessed after a break).

Reply:
> Ideally, we shouldn't throw ObjectDisposedException at all in this case (since a user didn't actually dispose anything), but that's a good enough solution.

I think there's precedence for objects transitioning to disposed state without the user explicitly calling Dispose (IIRC SqlClient does this too), so I don't think the current way is bad.. But anyway...

> The problem is, we don't dispose NpgsqlDataReader on break but close it (I guess it was done like this so NpgsqlDataReader will throw a correct exception whenever it's accessed after a break).

Makes sense... since DbDataReader has a closed state which is distinct from disposed. There's also e.g. the COPY binary streams, but in any case I think we're pretty good as we are.